src/main/java/com/example/demo/categories/CategoriesService.java

```diff
@@ -10,12 +10,12 @@
 public class CategoriesService {
 
 	@Autowired
-	private CategoriesDao categoriesDao;
+	private CategoriesRepository categoriesRepository;
 	
 	
 	@Transactional
 	public Categories getCategory(String element) {
-		Categories category = categoriesDao.getCategory(element);
+		Categories category = categoriesRepository.getCategory(element);
```

Comment:
```suggestion
		return categoriesRepository.getCategory(element);
```
Nie łatwiej tak? ;) + nie alokujemy zmiennej w pamięci